pkg/skaffold/parser/configlocations/configlocations.go

```diff
@@ -0,0 +1,309 @@
+	"path"
+	"reflect"
+	"strconv"
+	"strings"
+
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/output/log"
+	sErrors "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/errors"
+	latestV1 "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest/v1"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+type YAMLInfo struct {
+	RNode      *kyaml.RNode
+	SourceFile string
+}
+
+type Location struct {
+	SourceFile  string
+	StartLine   int
+	StartColumn int
+	EndLine     int
+	EndColumn   int
+}
+
+type YAMLInfos struct {
+	// TODO(aaron-prindle)make private again once Parse is setup properly
+	YamlInfos map[uintptr]map[string]YAMLInfo
+	// rename FieldsOverrodeByProfile -> SchemaPathsOverrodeByProfile
+	FieldsOverrodeByProfile map[string]YAMLOverrideInfo // map of schema path -> profile name
+}
+
+func MissingLocation() *Location {
+	return &Location{
+		SourceFile:  "",
+		StartLine:   -1,
+		StartColumn: -1,
+		EndLine:     -1,
+		EndColumn:   -1,
+	}
+}
+
+func NewYAMLInfos() *YAMLInfos {
+	return &YAMLInfos{
+		YamlInfos: map[uintptr]map[string]YAMLInfo{},
+	}
+}
+
+type YAMLOverrideInfo struct {
+	ProfileName    string
+	PatchIndex     int
+	PatchOperation string
+	PatchCopyFrom  string
+}
+
+// Parse parses a skaffold config entry collecting file location information for each schema config object
+func Parse(sourceFile string, config *latestV1.SkaffoldConfig, fieldsOverrodeByProfile map[string]YAMLOverrideInfo) (*YAMLInfos, error) {
+	yamlInfos, err := buildMapOfSchemaObjPointerToYAMLInfos(sourceFile, config, map[uintptr]map[string]YAMLInfo{}, fieldsOverrodeByProfile)
+	return &YAMLInfos{
+			YamlInfos:               yamlInfos,
+			FieldsOverrodeByProfile: fieldsOverrodeByProfile,
+		},
+		err
+}
+
+// Locate gets the location for a skaffold schema struct pointer
+func (m *YAMLInfos) Locate(obj interface{}) *Location {
+	return m.locate(obj, "")
+}
+
+// Locate gets the location for a skaffold schema struct pointer
+func (m *YAMLInfos) LocateElement(obj interface{}, idx int) *Location {
+	return m.locate(obj, strconv.Itoa(idx))
+}
+
+// Locate gets the location for a skaffold schema struct pointer
+func (m *YAMLInfos) LocateField(obj interface{}, fieldName string) *Location {
+	return m.locate(obj, fieldName)
+}
+
+func (m *YAMLInfos) locate(obj interface{}, key string) *Location {
+	kind := reflect.ValueOf(obj).Kind()
+	if kind != reflect.Ptr {
+		log.Entry(context.TODO()).Infof("non pointer object passed to Locate: %v of type %T", obj, obj)
+		return MissingLocation()
+	}
+	if _, ok := m.YamlInfos[reflect.ValueOf(obj).Pointer()]; !ok {
+		log.Entry(context.TODO()).Infof("no map entry found when attempting Locate for %v of type %T and pointer: %d", obj, obj, reflect.ValueOf(obj).Pointer())
+		return MissingLocation()
+	}
+	node := m.YamlInfos[reflect.ValueOf(obj).Pointer()][key]
+	if _, ok := m.YamlInfos[reflect.ValueOf(obj).Pointer()][key]; !ok {
```

Comment:
i think we don't need to do 2 reads here
```suggestion
	node, ok := m.YamlInfos[reflect.ValueOf(obj).Pointer()][key]
	if !ok {
```

Reply:
Good catch, done